pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -89,7 +89,13 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 			break
 		}
 		if i == 3 {
-			return cluster, fmt.Errorf("failed to remove the cluster [%s]: %v", cluster.Name, err)
+			if apierrors.IsNotFound(err) {
+				logrus.Warnf("%v", err)
```

Comment:
add explanation aroudn the warn. dont just log ghe error without any context

Reply:
Moved this logic to driver.go/driverRemove to be closer to the source of the api error.